src/formatting/formatter/datetime.rs

```diff
@@ -108,20 +159,27 @@ impl Formatter for DatetimeFormatter {
             datetime: DateTime<T>,
         ) -> Result<String, FormatError>
         where
-            T: TimeZone,
+            T: TimeZone + TimezoneName,
             T::Offset: Display,
         {
             Ok(match this {
-                DatetimeFormatter::Chrono { items, locale } => match *locale {
-                    Some(locale) => datetime.format_localized_with_items(items.iter(), locale),
-                    None => datetime.format_with_items(items.iter()),
+                DatetimeFormatter::Chrono { items, locale } => {
+                    let new_items = items.iter().map(|item| match item {
+                        Item::Fixed(Fixed::TimezoneName) => T::timezone_name(&datetime),
+                        item => item.to_owned(),
```

Comment:
Do you think it's worth avoiding some allocations by using something like this here?

```rust
fn borrow_item<'a>(item: &'a Item) -> Item<'a> {
    match item {
        Item::Literal(s) => Item::Literal(s),
        Item::Space(s) => Item::Space(s),
        Item::Numeric(n, p) => Item::Numeric(n.clone(), *p),
        Item::Fixed(f) => Item::Fixed(f.clone()),
        Item::OwnedLiteral(l) => Item::Literal(&*l),
        Item::OwnedSpace(s) => Item::Space(&*s),
        Item::Error => Item::Error,
    }
}
```